app/models/info_request.rb

```diff
@@ -1739,6 +1739,12 @@ def holding_pen_request?
     self == self.class.holding_pen_request
   end
 
+  def all_citations
```

Comment:
How about this to keep it cleaner?

```diff
diff --git a/app/models/citation.rb b/app/models/citation.rb
index e3ecf9f50..473d421aa 100644
--- a/app/models/citation.rb
+++ b/app/models/citation.rb
@@ -30,6 +30,11 @@ class Citation < ApplicationRecord
   validates :type, presence: true,
                    inclusion: { in: %w(news_story academic_paper other) }

+  scope :for_request, ->(info_request) do
+    where(citable: info_request).
+      or(where(citable: info_request.info_request_batch))
+  end
+
   def applies_to_batch_request?
     citable.is_a?(InfoRequestBatch)
   end
diff --git a/app/models/info_request.rb b/app/models/info_request.rb
index ffd30900e..c660ae935 100644
--- a/app/models/info_request.rb
+++ b/app/models/info_request.rb
@@ -112,7 +112,7 @@ class InfoRequest < ApplicationRecord
            :inverse_of => :info_request,
            :dependent => :destroy
   has_many :citations,
-           -> { order('created_at') },
+           -> (info_request) { unscope(:where).for_request(info_request) },
            as: :citable,
            inverse_of: :citable,
            dependent: :destroy
```

(h/t https://stackoverflow.com/questions/53427787/how-to-override-the-query-in-has-many-relationship)

Reply:
And a counterpart for collecting all citations for a batch, plus all request-specific citations in that batch:

```diff
diff --git a/app/models/citation.rb b/app/models/citation.rb
index e3ecf9f50..40e58f3d4 100644
--- a/app/models/citation.rb
+++ b/app/models/citation.rb
@@ -30,6 +30,16 @@ class Citation < ApplicationRecord
   validates :type, presence: true,
                    inclusion: { in: %w(news_story academic_paper other) }

+  scope :for_batch, ->(info_request_batch) do
+    where(citable: info_request_batch).
+      or(where(citable: info_request_batch.info_requests))
+  end
+
   def applies_to_batch_request?
     citable.is_a?(InfoRequestBatch)
   end

```

Reply:
Yeah that was quick stub to see if the associations were working - happy to go with your suggestion